leptos-leaflet/src/components/quad_tile_layer.rs

```diff
@@ -0,0 +1,185 @@
+    let get_tile_url_closure: JsStoredValue<Option<Closure<dyn Fn(JsValue) -> String>>> = 
+        JsStoredValue::new_local(None);
+
+    Effect::new(move |_| {
+        if let Some(map) = map_context.map() {
+            // Create tile layer options
+            let options = leaflet::TileLayerOptions::default();
+            if !attribution.is_empty() {
+                options.set_attribution(attribution.to_string());
+            }
+            options.set_min_zoom(min_zoom);
+            options.set_max_zoom(max_zoom);
+            
+            // Create a standard tile layer first
+            let map_layer = leaflet::TileLayer::new_options(&url, &options);
+            
+            // Override the getTileUrl method to use quadkey
+            let url_pattern = url.clone();
+            let closure = Closure::wrap(Box::new(move |coords: JsValue| -> String {
+                // Extract x, y, z from coords object
+                let x = js_sys::Reflect::get(&coords, &JsValue::from_str("x"))
+                    .unwrap_or(JsValue::from(0))
+                    .as_f64()
+                    .unwrap_or(0.0) as u32;
+                let y = js_sys::Reflect::get(&coords, &JsValue::from_str("y"))
+                    .unwrap_or(JsValue::from(0))
+                    .as_f64()
+                    .unwrap_or(0.0) as u32;
+                let z = js_sys::Reflect::get(&coords, &JsValue::from_str("z"))
+                    .unwrap_or(JsValue::from(0))
+                    .as_f64()
+                    .unwrap_or(0.0) as u32;
+                
+                let quadkey = tile_to_quadkey(x, y, z);
+                url_pattern.replace("{q}", &quadkey)
+            }) as Box<dyn Fn(JsValue) -> String>);
+            
+            // Override the getTileUrl method on the layer instance
+            if let Err(e) = js_sys::Reflect::set(
+                &map_layer,
+                &JsValue::from_str("getTileUrl"),
+                closure.as_ref().unchecked_ref(),
+            ) {
+                warn!("Failed to set getTileUrl method: {:?}", e);
+                return;
+            }
+            
+            // Store the closure to prevent it from being dropped
+            get_tile_url_closure.set_value(Some(closure));
+            
+            map_layer.add_to(&map);
+
+            match (bring_to_front, bring_to_back) {
+                (true, true) => warn!("The parameters are set to bring the layer to front and back at the same time. Ignoring these parameters..."),
+                (true, false) => {map_layer.bring_to_front();}
+                (false, true) => {map_layer.bring_to_back();}
+                (false, false) => (),
+            }
+
+            let map_layer = JsStoredValue::new_local(map_layer);
+
+            on_cleanup(move || {
+                map_layer.with_value(|v| v.remove());
+                // Clean up the closure when the component is cleaned up
+                get_tile_url_closure.set_value(None);
```

Comment:
Double check whether this is really necessary. Usually all StoredValues should be cleaned up automatically when the component goes out of scope.